feat: generalized SCM terms (#91)
Legit-Labs/legitify: README.md

 <img width="250" alt="Legitify Logo" src="https://user-images.githubusercontent.com/74864790/174815311-746a0c98-9a1f-44a9-808c-035788edfd4d.png">
 

> Strengthen the security posture of your organization! <br/>

Reviewer: Strengthen the security posture of your organization! <br/>
```suggestion
Strengthen the security posture of your GitHub/GitLab servers <br/>
```

> Strengthen the security posture of your GitHub/GitLab servers <br/>
> Detect and remediate misconfigurations, security and compliance issues across all your code management platforms with ease 🔥 <br/>

Reviewer 2: ```suggestion
Strengthen the security posture of your source-code management! <br/>
Detect and remediate misconfigurations, security and compliance issues across all your GitHub and GitLab assets with ease 🔥 <br/>
```

2. legitify requires a GitHub personal access token (PAT) to analyze your resources successfully, which can be either provided as an argument (`-t`) or as an environment variable (`LEGITIFY_TOKEN`).
   The PAT needs the following scopes for full analysis:
  ```
  admin:org, read:enterprise, admin:org_hook, read:org, repo, read:repo_hook
  ```
See [Creating a Personal Access Token](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token) for more information.  
Fine-grained personal access tokens are currently not supported because they do not support GitHub's GraphQL (https://github.blog/2022-10-18-introducing-fine-grained-personal-access-tokens-for-github/)

### GitHub Enterprise Server
You can run legitify against a GitHub Enterprise Server instance if you set the endpoint URL in the environment variable ``SERVER_URL``:

```sh
export SERVER_URL="https://github.example.com/"
LEGITIFY_TOKEN=<your_token> legitify analyze --org org1,org2 --namespace organization,member
```
## GitLab Cloud/Server Support

> 1. As mentioned in the previous section, you need to be an owner of at least one GitLab group. Otherwise, you can still use the tool if you're an admin of at least one project inside a group, in which case you'll be able to see only repository-related policies results.

Reviewer 2: That's true for GL cloud. But for GL server, don't you need to be an admin on the instance?

Author: No, it's okay to be only an owner to provide the same level of collection

Reviewer 2: OK, we'll have to modify it when we'll add server-level policies

> ```

Reviewer 2: Under "## Usage", consider adding a brief description of the new "--scm" option

The above command will test organization and member policies against org1 and org2.

> ## GitHub Requirements

Reviewer 2: ```suggestion
## Requirements
### GitHub Cloud 
```

1. To get the most out of legitify, you need to be an owner of at least one GitHub organization. Otherwise, you can still use the tool if you're an admin of at least one repository inside an organization, in which case you'll be able to see only repository-related policies results.
2. legitify requires a GitHub personal access token (PAT) to analyze your resources successfully, which can be either provided as an argument (`-t`) or as an environment variable (`$GITHUB_ENV`).

>    The PAT needs the following scopes for full analysis:

Reviewer 2: We should indicate the equivalent information under GHES and GL sections

  ```
  admin:org, read:enterprise, admin:org_hook, read:org, repo, read:repo_hook
  ```
See [Creating a Personal Access Token](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token) for more information.  
Fine-grained personal access tokens are currently not supported because they do not support GitHub's GraphQL (https://github.blog/2022-10-18-introducing-fine-grained-personal-access-tokens-for-github/)

> ## GitHub Enterprise Support
> You can run legitify against a GitHub Enterprise instance if you set the endpoint URL in the environment variable ``SERVER_URL``:

Reviewer 2: ```suggestion
### GitHub Enterprise Server
You can run legitify against a GitHub Enterprise Server instance if you set the endpoint URL in the environment variable ``SERVER_URL``:
```

> 

Reviewer 2: Please change line 73 to: "### GitLab Cloud/Server"

## Namespaces
Namespaces in legitify are resources that are collected and run against the policies.
Currently, the following namespaces are supported:
1. `organization` - GitHub organization (or GitLab group) level policies (e.g., "Two-Factor Authentication Is Not Enforced for the Organization")
2. `actions`      - organization GitHub Actions policies (e.g., "GitHub Actions Runs Are Not Limited To Verified Actions")

> 3. `member`       - GitHub organization (or GitLab group) members policies (e.g., "Stale Admin Found")

Reviewer 2: ```suggestion
3. `member`       - contributor level policies (e.g., "Stale Admin Found")
```
